packages/frontend/src/redux/actions/staking.js

```diff
@@ -533,3 +524,85 @@ export const handleUpdateCurrent = (accountId) => async (dispatch, getState) =>
     let currentAccount = selectStakingCurrentAccountbyAccountId(getState(), { accountId });
     dispatch(staking.updateCurrent({ currentAccount }));
 };
+
+export const getValidatorFarmData = (validator, accountId) => async (dispatch, getState) => {
+
+    if (validator?.version !== FARMING_VALIDATOR_VERSION || !accountId) return;
+
+    const poolSummary = await validator.contract.get_pool_summary();
+    const farms = await validator.contract.get_farms({ from_index: 0, limit: 300 });
+
+    const list = await Promise.all(farms.map(({ token_id, farm_id, active }) => {
+        try {
+            dispatch(fetchToken({ contractName: token_id }));
```

Comment:
`await` should be here so that if this call throws an error, we fall into the `catch()` block before trying to call `get_unclaimed_reward` (if the token cannot be loaded, we won't know how to _format_ the unclaimed reward)